### PR TITLE
Confirmed Flight Paths

### PR DESCRIPTION
**Pull request by ServiceDog**
Need to track flight paths in postgis to determine intersection with other aircraft.

Features:
- [x] A new table "flights" that tracks
    - flight id
    - aircraft id
    - path (linestringZ)
    - isa (3D volume) 
        - simply the "envelope" of the path
    - time_start
    - time_end
- [x] Best path checks if a flight will intersect any existing flight paths (in addition to zone intersections)
    - Paths are broken into many smaller line segments with start/end times and compared for intersection.
    - e.g. Two flights have the same path with overlapping schedules, but one aircraft leaves 60 seconds earlier. The aircraft might never come within 30m of each other. With segmentized paths, aircraft A's first path segment might not overlap with aircraft B's first path segment because of the time separation, and so on
- [x] Auto-remove old flights from DB if time_end is in the past by more than N hours
    - svc-storage has long-term storage of flight details
- [x] Pull from redis queue to update flight paths instead of using gRPC interface

### CODE REVIEW

File: server/src/postgis/aircraft.rs [hunk 1]
```diff
@@ -295,16 +296,17 @@ pub async fn update_aircraft_position(aircraft: Vec<AircraftPosition>) -> Result
         PostgisError::Aircraft(AircraftError::DBError)
     })?;
 
+    let table_name = format!("{}.aircraft", super::PSQL_SCHEMA);
     let stmt = transaction
-        .prepare_cached(
+        .prepare_cached(&format!(
             "
-        INSERT INTO arrow.aircraft (identifier, geom, last_position_update)
+        INSERT INTO {table_name} (identifier, geom, last_position_update)
```
Comment (lotjuh) [inline server/src/postgis/aircraft.rs:303]: When using templated queries, it's saver to add the double quotes around the table and column names, making sure you're not invoking a function or using a reserved word.
You can use the r#"" notation to add double quotes without escaping them to keep it readable.
Example can be found in the [`svc-storage` code](https://github.com/Arrow-air/svc-storage/blob/133e4bbe57ba2b852797a74d6e4d3b8cc94a7539/server/src/postgres/queries.rs#L230)

File: server/src/postgis/aircraft.rs [hunk 2]
```diff
@@ -415,10 +417,11 @@ pub async fn update_aircraft_velocity(aircraft: Vec<AircraftVelocity>) -> Result
         PostgisError::Aircraft(AircraftError::DBError)
     })?;
 
+    let table_name = format!("{}.aircraft", super::PSQL_SCHEMA);
     let stmt = transaction
-        .prepare_cached(
+        .prepare_cached(&format!(
             "
-        INSERT INTO arrow.aircraft (
+        INSERT INTO {table_name} (
```
Comment (lotjuh) [inline server/src/postgis/aircraft.rs:424]: see comment above

File: server/src/postgis/aircraft.rs [hunk 3]
```diff
@@ -491,7 +494,8 @@ pub async fn update_aircraft_velocity(aircraft: Vec<AircraftVelocity>) -> Result
 
 /// Gets the geometry of an aircraft given its identifier.
 pub async fn get_aircraft_pointz(identifier: &str) -> Result<PointZ, PostgisError> {
-    let stmt = "SELECT geom FROM arrow.aircraft WHERE identifier = $1;";
+    let table_name = format!("{}.aircraft", super::PSQL_SCHEMA);
+    let stmt = format!("SELECT geom FROM {table_name} WHERE identifier = $1;");
```
Comment (lotjuh) [inline server/src/postgis/aircraft.rs:498]: see comment above

File: server/src/postgis/waypoint.rs
Comment (lotjuh) [inline server/src/postgis/waypoint.rs:1]: Add double quotes for schema and column names in queries

File: server/src/postgis/vertiport.rs
Comment (lotjuh) [inline server/src/postgis/vertiport.rs:1]: Add double quotes for schema and column names in queries

File: server/src/main.rs [hunk 4]
```diff
@@ -56,6 +62,12 @@ async fn main() -> Result<(), Box<dyn std::error::Error>> {
 
     postgis::psql_init().await?;
 
+    // Start the maintenance threads
+    if psql_maintenance().await.is_err() {
```
[the remark this answers is not in the document]
Comment (ServiceDog) [reply server/src/main.rs:66]: Sounds good, will remove the maintenance thread :)

[the remark this answers is not in the document]
Comment (ServiceDog) [reply server/src/main.rs]: https://github.com/Arrow-air/svc-gis/pull/18/commits/0846dfb6de39162bef367ca775ff602305a8b0fd

File: server/src/postgis/aircraft.rs (hunk 1 above)
Comment (ServiceDog) [reply server/src/postgis/aircraft.rs]: https://github.com/Arrow-air/svc-gis/pull/18/commits/0846dfb6de39162bef367ca775ff602305a8b0fd

File: server/src/postgis/aircraft.rs (hunk 2 above)
Comment (ServiceDog) [reply server/src/postgis/aircraft.rs:436]: https://github.com/Arrow-air/svc-gis/pull/18/commits/0846dfb6de39162bef367ca775ff602305a8b0fd

File: server/src/postgis/aircraft.rs (hunk 3 above)
Comment (ServiceDog) [reply server/src/postgis/aircraft.rs]: https://github.com/Arrow-air/svc-gis/pull/18/commits/0846dfb6de39162bef367ca775ff602305a8b0fd

File: server/src/postgis/waypoint.rs
Comment (ServiceDog) [reply server/src/postgis/waypoint.rs:1]: https://github.com/Arrow-air/svc-gis/pull/18/commits/0846dfb6de39162bef367ca775ff602305a8b0fd

File: server/src/postgis/vertiport.rs
Comment (ServiceDog) [reply server/src/postgis/vertiport.rs:1]: https://github.com/Arrow-air/svc-gis/pull/18/commits/0846dfb6de39162bef367ca775ff602305a8b0fd

File: server/src/postgis/nearest.rs
[the remark this answers is not in the document]
Comment (ServiceDog) [reply server/src/postgis/nearest.rs:1]: https://github.com/Arrow-air/svc-gis/pull/18/commits/0846dfb6de39162bef367ca775ff602305a8b0fd

**Review by owlot — APPROVED** [review]

**Pull request merged**